### PR TITLE
docs: consistent remote location URLs + related copy edits

### PR DESCRIPTION
**Pull request by jorgeorpinel**
> Based on stuff I had stashed waiting for previous misc. updates PR to merge

- Remote URL revisions in multiple docs
- Misc. copy edits here and there

### CODE REVIEW

File: content/docs/command-reference/get-url.md [hunk 1]
```diff
@@ -37,8 +36,8 @@ DVC supports several types of (local or) remote locations (protocols):
 | Type    | Description    | `url` format                               |
 | ------- | -------------- | ------------------------------------------ |
 | `local` | Local path     | `/path/to/local/data`                      |
-| `s3`    | Amazon S3      | `s3://mybucket/data`                       |
-| `gs`    | Google Storage | `gs://mybucket/data`                       |
+| `s3`    | Amazon S3      | `s3://bucket/key`                          |
+| `gs`    | Google Storage | `gs://bucket/data`                         |
 | `ssh`   | SSH server     | `ssh://user@example.com:/path/to/data`     |
 | `hdfs`  | HDFS to file\* | `hdfs://user@example.com/path/to/data.csv` |
 | `http`  | HTTP to file\* | `https://example.com/path/to/data.csv`     |
```
Comment (jorgeorpinel) [inline content/docs/command-reference/get-url.md:43]: I noticed `arure`, `gdrive`, and `remote://` are missing here. Hmmm

File: content/docs/command-reference/import-url.md [hunk 2]
```diff
@@ -59,13 +58,13 @@ DVC supports several types of (local or) remote locations (protocols):
...
-| `azure`  | Microsoft Azure Blob Storage                        | `azure://my-container-name/path/to/data`   |
-| `gs`     | Google Cloud Storage                                | `gs://mybucket/data`                       |
+| `s3`     | Amazon S3                                           | `s3://bucket/data`                         |
+| `azure`  | Microsoft Azure Blob Storage                        | `azure://container/path/to/data`           |
+| `gs`     | Google Cloud Storage                                | `gs://bucket/data`                         |
 | `ssh`    | SSH server                                          | `ssh://user@example.com:/path/to/data`     |
 | `hdfs`   | HDFS to file (explanation below)                    | `hdfs://user@example.com/path/to/data.csv` |
 | `http`   | HTTP to file with _strong ETag_ (explanation below) | `https://example.com/path/to/data.csv`     |
-| `remote` | Remote path (see explanation below)                 | `remote://myremote/path/to/data`           |
+| `remote` | Remote path (see explanation below)                 | `remote://remote-name/path/to/data`        |
```
Comment (jorgeorpinel) [inline content/docs/command-reference/import-url.md:67]: This list i sonly missing `gdrive://`

File: content/docs/command-reference/get-url.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/get-url.md:43]: Could also reorder these to match remote refs and external data guides.

File: content/docs/command-reference/get-url.md [hunk 3]
```diff
@@ -34,23 +33,34 @@ directory will be placed inside.
...
+| `s3`     | Amazon S3                    | `s3://bucket/key/to/data`                              |
+| `azure`  | Microsoft Azure Blob Storage | `azure://container/path/to/data`                       |
+| `gdrive` | Google Drive                 | `gdrive://<folder-id>/data`                            |
+| `gs`     | Google Cloud Storage         | `gs://bucket/path/to/data`                             |
+| `ssh`    | SSH server                   | `ssh://user@example.com:/path/to/data`                 |
+| `hdfs`   | HDFS to file\*               | `hdfs://user@example.com/path/to/data.csv`             |
+| `http`   | HTTP to file\*               | `https://example.com/path/to/data.csv`                 |
+| `webdav` | WebDav to file\*             | `webdavs://example.com/public.php/webdav/path/to/data` |
+| `local`  | Local path                   | `/path/to/local/data`                                  |
+| `remote` | Remote path\*                | `remote://remote-name/path/to/data`                    |
```
Comment (jorgeorpinel) [inline content/docs/command-reference/get-url.md:47]: Is this list correct? does `get-url` support `remote://` for example? Maybe there was a reason it wasn't included before. Cc @efiop @shcheklein thanks

Comment (jorgeorpinel) [reply content/docs/command-reference/get-url.md:47]: Oh and should OSS be included too? Or maybe just mentioned as part of S3

File: content/docs/command-reference/import-url.md [hunk 4]
```diff
@@ -56,27 +55,29 @@ source.
...
+| `s3`     | Amazon S3                         | `s3://bucket/key/to/data`                              |
+| `azure`  | Microsoft Azure Blob Storage      | `azure://container/path/to/data`                       |
+| `gdrive` | Google Drive                      | `gdrive://<folder-id>/data`                            |
+| `gs`     | Google Cloud Storage              | `gs://bucket/path/to/data`                             |
+| `ssh`    | SSH server                        | `ssh://user@example.com:/path/to/data`                 |
+| `hdfs`   | HDFS to file\*                    | `hdfs://user@example.com/path/to/data.csv`             |
+| `http`   | HTTP to file with _strong ETag_\* | `https://example.com/path/to/data.csv`                 |
+| `webdav` | WebDav to file\*                  | `webdavs://example.com/public.php/webdav/path/to/data` |
+| `local`  | Local path                        | `/path/to/local/data`                                  |
+| `remote` | Remote path\*                     | `remote://remote-name/path/to/data`                    |
```
Comment (jorgeorpinel) [inline content/docs/command-reference/import-url.md:69]: Same Qs. Added GDrive and WebDAV so far.

**Comment by jorgeorpinel** [comment]
Also, the examples in https://dvc.org/doc/user-guide/external-dependencies and https://dvc.org/doc/user-guide/managing-external-data are also missing a bunch of remote types. Should we include them all like in get/import-url refs above?

**Comment by shcheklein** [comment]
@jorgeorpinel PR is getting out of hands. Let's split the scope and changes please.

File: content/docs/command-reference/config.md [hunk 5]
```diff
@@ -205,7 +205,7 @@ to learn more about the state file (database) that is used for optimization.
 > [Create a Bucket](https://docs.aws.amazon.com/AmazonS3/latest/gsg/CreatingABucket.html).
 
 ```dvc
-$ dvc remote add myremote s3://bucket/path
+$ dvc remote add myremote s3://bucket/key
```
Comment (shcheklein) [inline content/docs/command-reference/config.md:208]: why do we change this? it's confusing for most people.

File: content/docs/command-reference/get-url.md (hunk 3 above)
Comment (shcheklein) [reply content/docs/command-reference/get-url.md:47]: I guess we are overcomplicating this a bit ... what was the problem with all previous path URLs we had?

File: content/docs/command-reference/import-url.md (hunk 4 above)
Comment (shcheklein) [reply content/docs/command-reference/import-url.md:69]: it's hard to review this ... it's hard to understand what is changed exactly? could you summarize it please?

File: content/docs/user-guide/external-dependencies.md [hunk 6]
```diff
@@ -72,12 +54,12 @@ $ dvc run -n download_file
 
 ```dvc
 $ dvc run -n download_file
-          -d azure://my-container-name/data.txt \
```
Comment (shcheklein) [inline content/docs/user-guide/external-dependencies.md:75]: @jorgeorpinel do we really need updates like this? can we at least isolate and do them in a separate PR - so that it's clear - nothing to review, just merge

File: content/docs/user-guide/managing-external-data.md [hunk 7]
```diff
@@ -142,3 +126,25 @@ $ dvc run -d data.txt \
 
 Note that as long as there is a `hdfs://...` path for your data, DVC can handle
 it. So systems like Hadoop, Hive, and HBase are supported!
+
+### Local file system path
+
+The default cache location is `.dvc/cache`, so there is no need to move it for
```
Comment (shcheklein) [inline content/docs/user-guide/managing-external-data.md:132]: not sure I understand this change. let's do this as a separate PR with some scope and context please

File: content/docs/command-reference/get-url.md (hunk 3 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/get-url.md:47]: With these particular lists the main problem was that they were incomplete (still missing OSS).

File: content/docs/user-guide/external-dependencies.md (hunk 6 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md:75]: Yes, I can split it. This PR is a draft. I try to check them myself before requesting a review.

Please keep in mind that I put all the misc. updates into this `jorge` branch, and they mount up quickly, so it's not easy to balance the size of these generic PRs vs. the number of specialized branches and PRs.

File: content/docs/command-reference/get-url.md (hunk 3 above)
Comment (efiop) [reply content/docs/command-reference/get-url.md:47]: No, get-url doesn't support remote://

Comment (jorgeorpinel) [reply content/docs/command-reference/get-url.md:47]: Ah yeah that makes sense dah me 🤦 will remove it 🙂

File: content/docs/command-reference/config.md (hunk 5 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/config.md:208]: Ok, rolled these back. Maybe it should be `path/to/data` or `path/to/object` depending on the type I think.

File: content/docs/user-guide/external-dependencies.md (hunk 6 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/external-dependencies.md:57]: So going back to this. This PR's main thing is "Remote URL revisions in multiple docs" as mentioned in the description. This includes consistency in these sample hosts and paths, that only uses `-` in a minority of the cases, so I changed them to no `-`. Again, that's what this PR is mainly about (see other similar changes in the remote add ref).

I will try to extract other things not related to this idea before I request a review.

Thanks

File: content/docs/command-reference/get-url.md [hunk 8]
```diff
@@ -12,9 +12,8 @@ Download a file or directory from a supported URL (for example `s3://`,
 usage: dvc get-url [-h] [-q | -v] url [out]
 
 positional arguments:
-  url                   Location of the data to download.
-                        See supported URLs below.
-  out                   Destination path to put files in
+  url            (See supported URLs in the description.)
+  out            Destination path to put files in.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/get-url.md:16]: Std. way `url` argument is described in other ref Synopses.

File: content/docs/command-reference/get-url.md [hunk 9]
```diff
@@ -34,23 +33,33 @@ directory will be placed inside.
...
+| -------- | ---------------------------- | ---------------------------------------------------------- |
+| `s3`     | Amazon S3                    | `s3://bucket/data`                                         |
+| `azure`  | Microsoft Azure Blob Storage | `azure://container/data`                                   |
+| `gdrive` | Google Drive                 | `gdrive://<folder-id>/data`                                |
+| `gs`     | Google Cloud Storage         | `gs://bucket/data`                                         |
+| `ssh`    | SSH server                   | `ssh://user@example.com/path/to/data`                      |
+| `hdfs`   | HDFS to file\*               | `hdfs://user@example.com/path/to/data.csv`                 |
+| `http`   | HTTP to file\*               | `https://example.com/path/to/data.csv`                     |
+| `webdav` | WebDav to file\*             | `webdavs://example.com/public.php/webdav/path/to/data.csv` |
+| `local`  | Local path                   | `/path/to/local/data`                                      |
```
Comment (jorgeorpinel) [inline content/docs/command-reference/get-url.md:46]: Reordered list same as in remote add.modify refs. and added missing types (except OSS — not sure it's worth it).

Comment (jorgeorpinel) [reply content/docs/command-reference/get-url.md:46]: + improved the way notes about this list are described (below: lines 54+)

File: content/docs/command-reference/get-url.md [hunk 10]
```diff
@@ -73,19 +82,6 @@ $ wget https://example.com/path/to/data.csv
...
-```
-
-The above command will copy the `/local/path/to/data` file or directory into
-`./dir`.
-
-</details>
-
-<details>
-
 ### Click for Amazon S3 example
```
Comment (jorgeorpinel) [inline content/docs/command-reference/get-url.md:85]: Reordered this local example same as in remote add/modify refs (put in the bottom).

File: content/docs/command-reference/get-url.md (hunk 8 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/get-url.md:16]: Same in import-url.

File: content/docs/command-reference/get-url.md (hunk 9 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/get-url.md:46]: Same in import-url.

File: content/docs/command-reference/get-url.md (hunk 8 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/get-url.md:16]: And remote add

File: content/docs/command-reference/config.md (hunk 5 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/config.md]: p.s. There were instances of `bucket/mykey` before too, so given this comment I changed those to `path` also (88d5f270).

File: content/docs/command-reference/remote/add.md [hunk 11]
```diff
@@ -134,7 +133,7 @@ configure the remote's `endpointurl` explicitly:
 For example:
 
 ```dvc
-$ dvc remote add -d myremote s3://my-bucket/path/to/dir
+$ dvc remote add -d myremote s3://mybucket/path/to/dir
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/add.md:136]: Cosmetic: I removed `-` in URLs in this file and in remote modify, mainly.

File: content/docs/command-reference/remote/index.md [hunk 12]
```diff
@@ -103,7 +103,7 @@ remote = myremote
 > [Create a Bucket](https://docs.aws.amazon.com/AmazonS3/latest/gsg/CreatingABucket.html).
 
 ```dvc
-$ dvc remote add newremote s3://mybucket/myproject
+$ dvc remote add newremote s3://mybucket/path
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/index.md:106]: Changed possibly confusing concepts like "project" or "key" to a more generic "path" in URLs (mostly S3 URLs had this problem).

File: content/docs/user-guide/managing-external-data.md (hunk 7 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/managing-external-data.md]: Rolled back.

**Comment by jorgeorpinel** [comment]
I renamed the PR and focused the changes (down to 10 files from 16) @shcheklein. I also left a bunch of comments on what the changes are about startgin in https://github.com/iterative/dvc.org/pull/1695#pullrequestreview-472077874. PTAL

File: content/docs/command-reference/get-url.md [hunk 13]
```diff
@@ -34,23 +33,33 @@ directory will be placed inside.
...
+| Type     | Description                  | `url` format example                                       |
+| -------- | ---------------------------- | ---------------------------------------------------------- |
+| `s3`     | Amazon S3                    | `s3://bucket/data`                                         |
+| `azure`  | Microsoft Azure Blob Storage | `azure://container/data`                                   |
+| `gdrive` | Google Drive                 | `gdrive://<folder-id>/data`                                |
+| `gs`     | Google Cloud Storage         | `gs://bucket/data`                                         |
+| `ssh`    | SSH server                   | `ssh://user@example.com/path/to/data`                      |
+| `hdfs`   | HDFS to file\*               | `hdfs://user@example.com/path/to/data.csv`                 |
+| `http`   | HTTP to file\*               | `https://example.com/path/to/data.csv`                     |
+| `webdav` | WebDav to file\*             | `webdavs://example.com/public.php/webdav/path/to/data.csv` |
```
Comment (shcheklein) [inline content/docs/command-reference/get-url.md:45]: is `public.php` is a very common thingy for webdav?

File: content/docs/command-reference/remote/index.md (hunk 12 above)
Comment (shcheklein) [reply content/docs/command-reference/remote/index.md:106]: since we change this - we might want to use `-d` ?

File: content/docs/command-reference/get-url.md (hunk 13 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/get-url.md:45]: I'm not familiar TBH but apparently yes there's some sort of endpoint. Not sure if it's typically PHP. I took this from https://dvc.org/doc/command-reference/remote/add#supported-storage-types

But let me check, maybe it's quick to figure this out. ⌛

File: content/docs/command-reference/remote/index.md (hunk 12 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/index.md:106]: Probably not in this one under the **Customize an additional S3 remote** example (meant as a continuation of the previous example where a default local remote is added).

File: content/docs/command-reference/get-url.md (hunk 13 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/get-url.md:45]: From https://linuxconfig.org/webdav-server-setup-on-ubuntu-linux and https://docs.microsoft.com/en-us/iis/install/installing-publishing-technologies/installing-and-configuring-webdav-on-iis WebDAV is setup similar to any HTTP server so no special use of PHP here. We should probably review all these in a separate PR though, so merging this for now (will create a ticket).

Comment (skshetry) [reply content/docs/command-reference/get-url.md:45]: `public.php` is a public endpoint for Nextcloud and Owncloud. There's `remote.php` as a private endpoint. But, it can be removed from most of the places, but good to keep one as an example, as most of our WebDAV users are either using Owncloud or Nextcloud.
